extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -185,6 +196,20 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
       productForPdp = shoppingDataPerPage[0];
     }
 
+    // pdp view that calls the analytics service sending the product id
+    if (productForPdp) {
+      this.variableService_.onVariableUpdate(
+        AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+        productForPdp.productId
+      );
+    }
+
```

Comment:
Please add a comment her as well since it's a different concept than the previous block.

nit: I would move the analytics logic to the end of this function so it's not in the middle of the template logic.